__tests__/commands/upgrade.js

```diff
@@ -292,3 +292,16 @@ test.concurrent('can prune the offline mirror', (): Promise<void> => {
     expect(await fs.exists(path.join(config.cwd, `${mirrorPath}/dep-b-1.0.0.tgz`))).toEqual(false);
   });
 });
+
+test.concurrent('respects --scope flag', (): Promise<void> => {
+  return runUpgrade([], {scope: '@angular'}, 'respects-scope-flag', async (config): ?Promise<void> => {
+    const lockfile = explodeLockfile(await fs.readFile(path.join(config.cwd, 'yarn.lock')));
+    const pkg = await fs.readJson(path.join(config.cwd, 'package.json'));
+
+    expect(lockfile.indexOf('@angular/core@^2.4.9:')).toBeGreaterThanOrEqual(0);
+    expect(lockfile.indexOf('left-pad@^1.0.0:')).toBeGreaterThanOrEqual(0);
+
+    expect(pkg.dependencies['@angular/core']).not.toEqual('^2.4.9');
+    expect(pkg.dependencies['left-pad']).toEqual('^1.0.0');
+  });
+});
```

Comment:
Please also add a test to make sure that `-S @angular` doesn't upgrade `@angular-whatever`